[
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 2.118s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568880298.009900",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 08:14:00,812 INFO New Trending 2019-09-19\n{\"name\": \"Worn Powerstone\", \"s\": [0, 47, 33, 51, 62, 65, 73], \"z\": 1.2853538867878913}\n{\"name\": \"Death Baron\", \"s\": [0, 6, 7, 8, 17, 17, 14], \"z\": 1.2521514145040482}\n{\"name\": \"The Locust God\", \"s\": [0, 6, 8, 12, 18, 15, 18], \"z\": 1.22023267230613}\n{\"name\": \"Ob Nixilis, the Hate-Twisted\", \"s\": [0, 3, 2, 5, 13, 9, 11], \"z\": 1.2088093074390402}\n{\"name\": \"Stirring Wildwood\", \"s\": [0, 7, 3, 6, 12, 15, 13], \"z\": 1.2048537894620328}\n{\"name\": \"Animate Dead\", \"s\": [0, 47, 37, 50, 69, 50, 82], \"z\": 1.2043561391189452}\n{\"name\": \"Lord of the Accursed\", \"s\": [0, 5, 6, 7, 13, 12, 12], \"z\": 1.2041607355306472}\n{\"name\": \"Mikokoro, Center of the Sea\", \"s\": [0, 9, 3, 9, 17, 19, 13], \"z\": 1.198917463373708}\n{\"name\": \"Mind Stone\", \"s\": [0, 109, 107, 134, 158, 139, 148], \"z\": 1.1980735952565942}\n{\"name\": \"Plague Belcher\", \"s\": [0, 3, 4, 5, 11, 13, 9], \"z\": 1.19311518618657}\n2019-09-19 08:14:00,827 INFO daily.py took 0:13:56.004431\n```",
        "ts": "1568880836.010000",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 08:14:02,092 INFO backup.py took 0:14:00.031270\n```",
        "ts": "1568880837.010100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 1.135s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568887498.010200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 12:03:52,599 INFO cards.py took 0:03:46.999775\n```",
        "ts": "1568894628.010300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "Nightly started, onlydecks=False, skipdecks=False",
        "ts": "1568894636.010400",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 1.651s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568894697.010500",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 2.561s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568901899.010600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 2.286s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568909099.010700",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "Nightly completed",
        "ts": "1568909269.010800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 16:07:54,740 INFO nightly.py took 4:03:53.181454\n```",
        "ts": "1568909270.010900",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 16:36:11,672 ERROR Status code 404 from <https:\/\/www.youtube.com\/feeds\/videos.xml?playlist_id=PL-XQnS2UM7iJQmJJI2U9garQ8wTww4XqB>\n```",
        "ts": "1568910966.011000",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 31.478s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568916327.011100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 19:17:22,305 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;stdin&gt;\", line 1, in &lt;module&gt;\nTypeError: dump() missing 1 required positional argument: 'fp'\n```",
        "ts": "1568920637.011200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 19:53:43,441 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"\/usr\/lib\/python3.5\/code.py\", line 91, in runcode\n    exec(code, self.locals)\n  File \"&lt;console&gt;\", line 1, in &lt;module&gt;\nNameError: name 'all_carda' is not defined\n```",
        "ts": "1568922818.011300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 19:53:56,334 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"\/usr\/lib\/python3.5\/code.py\", line 91, in runcode\n    exec(code, self.locals)\n  File \"&lt;console&gt;\", line 1, in &lt;module&gt;\nKeyError: 'language'\n```",
        "ts": "1568922831.011400",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:00:53,013 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \".\/cards.py\", line 737, in &lt;module&gt;\n    _init(fast)\n  File \".\/cards.py\", line 44, in _init\n    _init_allcards(fast=fast)\n  File \".\/cards.py\", line 169, in _init_allcards\n    card['image_uris']['en'] = card['image_uris']\nKeyError: 'image_uris'\n```",
        "ts": "1568923248.011500",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:01:50,076 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;stdin&gt;\", line 1, in &lt;module&gt;\nTypeError: 'module' object is not subscriptable\n```",
        "ts": "1568923305.011600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:03:16,836 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \".\/cards.py\", line 740, in &lt;module&gt;\n    _init(fast)\n  File \".\/cards.py\", line 44, in _init\n    _init_allcards(fast=fast)\n  File \".\/cards.py\", line 426, in _init_allcards\n    cache.set('cards', allcards_dict)\n  File \"\/home\/ubuntu\/edhrec\/core\/cache.py\", line 89, in set\n    (key, value, expires)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/psycopg2\/extras.py\", line 141, in execute\n    return super(DictCursor, self).execute(query, vars)\npsycopg2.ProgrammingError: can't adapt type 'dict'\n```",
        "ts": "1568923392.011700",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 1.451s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568923498.011800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:09:39,359 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"\/usr\/lib\/python3.5\/code.py\", line 91, in runcode\n    exec(code, self.locals)\n  File \"&lt;console&gt;\", line 1, in &lt;module&gt;\n  File \"\/usr\/lib\/python3.5\/json\/__init__.py\", line 178, in dump\n    for chunk in iterable:\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 429, in _iterencode\n    yield from _iterencode_dict(o, _current_indent_level)\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 403, in _iterencode_dict\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 403, in _iterencode_dict\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 403, in _iterencode_dict\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 339, in _iterencode_dict\n    raise ValueError(\"Circular reference detected\")\nValueError: Circular reference detected\n```",
        "ts": "1568923774.011900",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:11:11,142 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"\/usr\/lib\/python3.5\/code.py\", line 91, in runcode\n    exec(code, self.locals)\n  File \"&lt;console&gt;\", line 1, in &lt;module&gt;\n  File \"\/usr\/lib\/python3.5\/json\/__init__.py\", line 178, in dump\n    for chunk in iterable:\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 429, in _iterencode\n    yield from _iterencode_dict(o, _current_indent_level)\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 403, in _iterencode_dict\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 403, in _iterencode_dict\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 324, in _iterencode_list\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 403, in _iterencode_dict\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 403, in _iterencode_dict\n    yield from chunks\n  File \"\/usr\/lib\/python3.5\/json\/encoder.py\", line 339, in _iterencode_dict\n    raise ValueError(\"Circular reference detected\")\nValueError: Circular reference detected\n```",
        "ts": "1568923866.012000",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:30:08,838 INFO archidekt.py took 0:25:44.512215\n```",
        "ts": "1568925004.012100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:34:23,468 INFO cards.py took 0:22:10.487109\n```",
        "ts": "1568925258.012200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:20:56:10] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\n  File \"\/home\/ubuntu\/edhrec\/api\/util.py\", line 31, in get_shit\n    image=card['image_uris'][lang]['normal'],\nKeyError: 'es'\n```",
        "ts": "1568926567.012300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 20:58:03,475 INFO deckstats.py took 0:25:54.611873\n```",
        "ts": "1568926678.012400",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:20:58:10] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\n  File \"\/home\/ubuntu\/edhrec\/api\/util.py\", line 32, in get_shit\n    isBanned=card['banned'],\nNameError: name 'language' is not defined\n```",
        "ts": "1568926685.012500",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:20:58:19] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\n  File \"\/home\/ubuntu\/edhrec\/api\/util.py\", line 32, in get_shit\n    isUnofficial=card['unofficial'],\nNameError: name 'language' is not defined\n```",
        "ts": "1568926694.012600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:20:58:25] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\n  File \"\/home\/ubuntu\/edhrec\/api\/util.py\", line 32, in get_shit\n    isUnofficial=card['unofficial'],\nNameError: name 'language' is not defined\n```",
        "ts": "1568926700.012700",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:20:58:30] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\n  File \"\/home\/ubuntu\/edhrec\/api\/util.py\", line 32, in get_shit\n    isUnofficial=card['unofficial'],\nNameError: name 'language' is not defined\n```",
        "ts": "1568926706.012800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "client_msg_id": "aadd6751-7fc3-4b42-b3a5-6523ea9e1262",
        "type": "message",
        "text": ".",
        "user": "U28TLDP0R",
        "ts": "1568926711.013000",
        "team": "T286HM2F3",
        "user_team": "T286HM2F3",
        "source_team": "T286HM2F3",
        "user_profile": {
            "avatar_hash": "2a61603e280e",
            "image_72": "https:\/\/avatars.slack-edge.com\/2016-11-16\/104915990673_2a61603e280ecee0ab38_72.jpg",
            "first_name": "David",
            "real_name": "David Keatts",
            "display_name": "keattz",
            "team": "T286HM2F3",
            "name": "keattz",
            "is_restricted": false,
            "is_ultra_restricted": false
        }
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:20:58:39] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\n  File \"\/home\/ubuntu\/edhrec\/api\/util.py\", line 32, in get_shit\n    isUnofficial=card['unofficial'],\nNameError: name 'language' is not defined\n```",
        "ts": "1568926714.013100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:21:00:26] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\nTypeError: get_shit() got an unexpected keyword argument 'lang'\n```",
        "ts": "1568926821.013200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[19\/Sep\/2019:21:00:47] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/jsontools.py\", line 58, in json_handler\n    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/api\/recs.py\", line 181, in index\n    card = get_shit(name, dir='cards', lang=lang)\nTypeError: get_shit() got an unexpected keyword argument 'lang'\n```",
        "ts": "1568926843.013300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "client_msg_id": "613e3fd7-8aa2-44e6-8f7a-8476dc0c93c8",
        "type": "message",
        "text": "hmm",
        "user": "U28TLDP0R",
        "ts": "1568926874.013500",
        "team": "T286HM2F3",
        "user_team": "T286HM2F3",
        "source_team": "T286HM2F3",
        "user_profile": {
            "avatar_hash": "2a61603e280e",
            "image_72": "https:\/\/avatars.slack-edge.com\/2016-11-16\/104915990673_2a61603e280ecee0ab38_72.jpg",
            "first_name": "David",
            "real_name": "David Keatts",
            "display_name": "keattz",
            "team": "T286HM2F3",
            "name": "keattz",
            "is_restricted": false,
            "is_ultra_restricted": false
        }
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 21:47:55,270 INFO mtggoldfish.py took 0:29:34.541753\n```",
        "ts": "1568929670.013600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 1.099s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568930697.013700",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-09-19 22:34:21,360 INFO update_deckswithprices.py took 0:30:05.341375\nNew deck count from past 2 days: \n- Archidekt: 725\n- Other: 0\n- MTGGoldfish: 407\n- deckstats: 500\n```",
        "ts": "1568932456.013800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 3.548s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568937899.013900",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 2.295s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568945098.014000",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 1.750s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568952297.014100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "\n```\nF.............................\n======================================================================\nFAIL: test_Atarka (__main__.testXenagos)\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/testing.py\", line 19, in test_Atarka\n    self.assertNotEqual(-1, data[bIndex:eIndex].find('Atarka'))\nAssertionError: -1 == -1\n\n----------------------------------------------------------------------\nRan 30 tests in 1.589s\n\nFAILED (failures=1)\n\n```",
        "ts": "1568959497.014200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/80588\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    }
]